function-sidecar/pkg/dispatcher/grpc/grpc.go

```diff
@@ -119,7 +119,7 @@ func NewGrpcDispatcher(port int, timeout time.Duration) (dispatcher.Dispatcher,
 		return nil, err
 	}
 
-	result := &grpcDispatcher{fnStream, make(chan message.Message, 100), make(chan message.Message, 100), make(chan bool, 10)}
+	result := &grpcDispatcher{fnStream, make(chan message.Message, 100), make(chan message.Message, 100), make(chan struct{}, 10)}
```

Comment:
Since the "done" channel can be signalled by closing it, there is no need for it to be buffered.